rpcserver.go

```diff
@@ -1934,28 +1934,70 @@ func (r *rpcServer) parseOpenChannelReq(in *lnrpc.OpenChannelRequest,
 
 	localFundingAmt := btcutil.Amount(in.LocalFundingAmount)
 	remoteInitialBalance := btcutil.Amount(in.PushSat)
-	minHtlcIn := lnwire.MilliSatoshi(in.MinHtlcMsat)
-	remoteCsvDelay := uint16(in.RemoteCsvDelay)
-	maxValue := lnwire.MilliSatoshi(in.RemoteMaxValueInFlightMsat)
-	maxHtlcs := uint16(in.RemoteMaxHtlcs)
-	remoteChanReserve := btcutil.Amount(in.RemoteChanReserveSat)
-
-	globalFeatureSet := r.server.featureMgr.Get(feature.SetNodeAnn)
 
-	// Ensure that a local funding amount has been specified.
-	if localFundingAmt == 0 {
+	// If we are not committing the maximum viable balance towards a channel
+	// then the local funding amount must be specified. In case FundMax is
+	// set the funding amount is specified as the interval between minimum
+	// funding amount and by the configured maximum channel size.
+	if !in.FundMax && localFundingAmt == 0 {
 		return nil, fmt.Errorf("local funding amount must be non-zero")
 	}
 
 	// Ensure that the initial balance of the remote party (if pushing
 	// satoshis) does not exceed the amount the local party has requested
-	// for funding.
-	//
-	if remoteInitialBalance >= localFundingAmt {
+	// for funding. This is only checked if we are not committing the
+	// maximum viable amount towards the channel balance. If we do commit
+	// the maximum then the remote balance is checked in a dedicated FundMax
```

Comment:
where is this check?

Reply:
Thanks for catching this! I thought this was checked here in the [rpcserver](https://github.com/lightningnetwork/lnd/pull/6903/files#diff-674c79ba02b6e9c45a994388392689814f3538800351b2fd43fb3ade21e1effbR1988).

I've threaded the push amount through to the `wallet_assembler` and added the [missing check](https://github.com/lightningnetwork/lnd/pull/6903/files#diff-ba25a5144f3090a192b11b77b5cbd5b63d1d33cfe61dd1e29cb15dabb27261e9R330) after the coin selection is done.

I've also added a [test](https://github.com/lightningnetwork/lnd/pull/6903/files#diff-91e9a84f98c8ad806da698b1a49266b069f77b5760196d5aed069e278fe1523aR120) to check this.